### ISSUE TITLE
Search for `go.mod` whenever if found a Go file

### ISSUE THREAD

**Issue by harryzcy** [super-linter/super-linter#5292]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Current Behavior

Go modules lints are run only when `go.mod` is included, not when only some Go files are included.

This is strange when `VALIDATE_ALL_CODEBASE` is set to `true`. A Go module does exist but it's not identified because `go.mod` is not updated.

### Expected Behavior

Go modules lints should run when `VALIDATE_ALL_CODEBASE: false` and some Go files are updated.

**Comment by ferrarimarco** [comment]
Hi @harryzcy ! Isn't this covered by `VALIDATE_GO` to `true` (the default)?

**Comment by harryzcy** [comment]
> Hi @harryzcy ! Isn't this covered by `VALIDATE_GO` to `true` (the default)?

`VALIDATE_GO_MODULES: true` has a lot more lint cases than `VALIDATE_GO`. I got many of those after upgrading super-linter to v6. And those lints only show up when `go.mod` gets changed.

**Comment by ferrarimarco** [comment]
@harryzcy this is not supported by golint. The last time I looked, it only supports the whole set of checks when linting Go modules, while only a subset of checks is enabled when linting file by file.

**Comment by ferrarimarco** [comment]
Ah,

**Comment by ferrarimarco** [comment]
I think I get what you mean. Linting of Go modules triggers when there are changes in `go.mod` only:

https://github.com/super-linter/super-linter/blob/d33107802b1ecab6ffc4146ac2d5644c72a673e4/lib/functions/buildFileList.sh#L316-L329

**Comment by harryzcy** [comment]
I mean I want to lint Go modules even when `go.mod` is not changed, but some Go files are changed.

**Comment by ferrarimarco** [comment]
Makes sense. I'll happily review a PR for this. We probably need a way to get the path to the `go.mod` file that is related to a Go file.
